app.R

```diff
@@ -532,11 +537,30 @@ server <- function(input, output) {
     # Require filtered_degInc data frame
     req(filtered_degInc())
     
-    fit <- lm(`Percentage by Total Income` ~ `Percentage by Generation`, data = filtered_degInc()) %>%
-          fitted.values()
+    # fit <- lm(`Percentage by Total Income` ~ `Percentage by Generation`, data = filtered_degInc()) %>%
+    #       fitted.values()
+    
+    # You can apply lm() with group_by() by using do()
+    # Source: https://github.com/tidyverse/dplyr/issues/2177
+    fit <- filtered_degInc() %>%
+      group_by(`Visible Minority`) %>%
+      do(lm(`Percentage by Total Income` ~ `Percentage by Generation`, data = .) %>%
+           coef() %>%
+           bind_rows()) %>%
+      ungroup()
+    
+    # The names from the lm() won't play nice with dplyr, so we rename them
+    names(fit) <- c("Visible Minority", "intercept", "slope")
+    
+    # In order to make sure that the graph has the same observations in the same
```

Comment:
The thing with `group_by` is that it doesn't preserve row order over the data frame. If you try to plot the lines with that, you just get scribbles.

By inner joining the model coefficients to the original dataset, we can make sure that the ordering is maintained.